cli/pcluster/utils.py

```diff
@@ -771,6 +771,17 @@ def get_info_for_amis(ami_ids):
         error(e.response.get("Error").get("Message"))
 
 
+def validate_pcluster_version_based_on_ami_name(ami_name):
+    match = re.match(r"(.*)aws-parallelcluster-(\d+\.\d+\.\d+)(.*)", ami_name)
+    if match:
+        if ami_name.find("aws-parallelcluster-{0}".format(get_installed_version())) < 0:
+            error(
```

Comment:
I am not sure if using error with `fail_on_error=True` is the behavior we want here. The bottom line is, in the case of cluster creation, we want to be able to turn this "name version check" off when `sanity_check = false`. Please double check to make sure this is the case.

Another way to handle this is we print out a warning instead of an error, so user is not forced to skip other sanity checks.

Reply:
On a second thought I think the current behavior is ok. I was worried that user with very customized AMI names might be affected, but I think the regex is fairly safe and they really should not have an older version in their AMI name.